ls1intum/Artemis
# Adding option to hide unreleased exercises to course exercise list

### Checklist

### Motivation and Context
Unreleased exercises are visible to tutors and instructors on the course exercise overview. Some tutors share their screen during tutor sessions and display the course exercise overview while navigating to the exercise they want to present. This leads to tutors "leaking" unreleased exercises on accident. To prevent this, tutors must be able to hide those exercises.
Adresses issues described in #2383 

### Description
Currently, it is possible to hide overdue exercises and exercises with full points. I have added a third checkbox that lets tutors and instructors hide unreleased exercises.

### Steps for Testing
You need: 1x Instructor 1x Tutor 1x Student 1x Test Course

1. Create a modeling exercise with a release date in the past
2. Create a modeling exercise and a quiz exercise with a release date in the future
3. As the student, navigate to the course exercise overview and make sure that
    - You **cannot** see the exercises with a release date in the future
    - You **cannot** see the checkbox "_Hide unreleased_"
4. As a tutor, navigate to the course exercise overview and make sure that
    - You **can** see the exercises with a release date in the future
    - You **can** see the checkbox "_Hide unreleased_"
    - The exercises with a release date in the future disappear if you check the checkbox and the exercise with a release date in 
      the past does not disappear
    - The exercises with a release date in the future appear again if you uncheck the checkbox 

### Screenshots
Without "Hide unreleased" enabled:
![num1](https://user-images.githubusercontent.com/45300855/115631815-da263380-a306-11eb-8a94-21ff97eb363e.JPG)

With "Hide unreleased" enabled:
![num2](https://user-images.githubusercontent.com/45300855/115631836-e3af9b80-a306-11eb-8077-6048f2367aad.JPG)
